relays/ethereum/src/utils.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+/// Macro that returns (client, Err(error)) tuple from function if result is Err(error).
+#[macro_export]
+macro_rules! bail_on_error {
```

Comment:
I'm pretty confused as to why we have so man functions that take a `Client` as an argument and also return one

Reply:
Because they need to consume `Client`? It was the requirement of the old `jsonrpsee` version (that has been used in the very beginning). On the last version(s), the client could be shared between several futures. So it can be changed. PRs are welcome :)

Reply:
Oh, I see an explanation of the `Client` thing. Fair enough, I've created an issue to rework this: https://github.com/paritytech/parity-bridges-common/issues/72